io/FV3GFS_io.F90

```diff
@@ -703,17 +704,24 @@ subroutine sfc_prop_restart_read (Sfcprop, Atm_block, Model, fv_domain)
         do ix = 1, Atm_block%blksz(nb)
           i = Atm_block%index(nb)%ii(ix) - isc + 1
           j = Atm_block%index(nb)%jj(ix) - jsc + 1
-          !--- assign hprime(1:10) and hprime(15:24) with new oro stat data
-          Sfcprop(nb)%hprime(ix,1)  = oro_ls_var(i,j,1)
-          Sfcprop(nb)%hprime(ix,2)  = oro_ls_var(i,j,2)
-          Sfcprop(nb)%hprime(ix,3)  = oro_ls_var(i,j,3)
-          Sfcprop(nb)%hprime(ix,4)  = oro_ls_var(i,j,4)
-          Sfcprop(nb)%hprime(ix,5)  = oro_ls_var(i,j,5)
-          Sfcprop(nb)%hprime(ix,6)  = oro_ls_var(i,j,6)
-          Sfcprop(nb)%hprime(ix,7)  = oro_ls_var(i,j,7)
-          Sfcprop(nb)%hprime(ix,8)  = oro_ls_var(i,j,8)
-          Sfcprop(nb)%hprime(ix,9)  = oro_ls_var(i,j,9)
-          Sfcprop(nb)%hprime(ix,10)  = oro_ls_var(i,j,10)
+          ! Replace hprime(1:10) with GSL oro stat data only when using GSL
+          ! drag suite with large scale GWD and blocking as part of unified drag
+          ! suite.   Otherwise, original oro stat data is used.
+          if ( (Model%gwd_opt==3 .or. Model%gwd_opt==33) .or.        &
```

Comment:
Aren't the first four if tests redundant, because you only end up in this block if the same tests equate to `.true.` in lines 661-662?

Does this block do anything if `Model%do_gsl_drag_ls_bl` is `.false.`? If not, then you can simply replace lines 661-662 with 710-712 and save yourself the if tests inside the do loop.

Reply:
The test on lines 710-711 are not redundant to the ones on lines 661-662.  Note the grouping of parentheses in lines 710-712:  the test is (3 .or. 33) .or. ((2 .or. 22) .and. do_gsl_drag_ls_bl).  In lines 661-662 it's (3 .or. 33 .or. 2 .or. 22).

If do_gsl_drag_ls_bl is .false. we still want the block (lines 710-724) to be executed if (3 .or. 33) is .true.

Reply:
Got it, sorry. This is so hard to see in Git diff view!